src/control/server/mgmt_svc_test.go

```diff
@@ -130,3 +141,174 @@ func TestCheckMgmtSvcReplica(t *testing.T) {
 		}
 	}
 }
+
+// mockDrpcClient is a mock of the DomainSocketClient interface
+type mockDrpcClient struct {
+	sync.Mutex
+	ConnectOutputError    error
```

Comment:
I have started using a pattern of separating the mock implementation from its configuration -- i.e. instead of making these configurable return values members of the mockDrpcClient struct, create a new mockDrpcClientConfig struct and use that for test configuration. In my experience it leads to cleaner test code and less churn over time because the mock initializers only need to accept the config rather than a zillion parameters which change every few weeks.

c.f. mocks in https://github.com/daos-stack/daos/blob/master/src/control/server/storage/scm/mocks.go

Note that the config member is not a pointer, which means that your mock implementation can return the zero values of the config struct if they're not otherwise defined.

Reply:
Good idea. I followed your advice and converted it.